### PR TITLE
TextUtils.join and .isEmpty changed to kotlin methods

### LINKED ISSUE #13036: cleanup: Replace most usages of `TextUtils` with Kotlin methods
We have a lot of usages of `TextUtils.isEmpty` and `TextUtils.join` that could be replaced with Kotlin methods, which are more idiomatic and have better contracts. Another reason to do it is to *perhaps* improve some tests performance, as said by `@mikehardy` [here](https://github.com/ankidroid/Anki-Android/pull/11684#discussion_r906317512).

These are quite simple to do, so it's fine to have many of these on a single PR.

As these are going to be pure simple non-functional changes, I'd rather keep this as a good first issue for ***only new contributors***.

So, if you are not a new contributor, your help would be welcome on any one of the other issues we have

### PR DESCRIPTION
**Pull request by soCallmeAdityaKumar**
## Purpose / Description
cleanup:Changed TextUtils.join to .jointostring and TextUtils.isEmpty to .isNullOrEmpty/isEmpty

## Fixes
Fixes #13036 

## Checklist
_Please, go through these checks before submitting the PR._

- [✓] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [✓] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 1]
```diff
@@ -2616,14 +2616,16 @@ open class CardBrowser :
                 withCol { changeDeckMulti(this, ids, newDid) }
             }
             if (result.succeeded()) {
+                Timber.d("Changed the SnackBar", "Snackbar")
```
Comment (criticalAY) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:2619]: Could you explain these changes?

File: AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ReviewerCustomFonts.kt [hunk 2]
```diff
@@ -79,9 +78,9 @@ class ReviewerCustomFonts(context: Context) {
     private fun getDominantFontStyle(context: Context, customFontsMap: Map<String?, AnkiFont>): String? {
         if (mDominantFontStyle == null) {
             mDominantFontStyle = getOverrideFontStyle(context, customFontsMap)
-            if (TextUtils.isEmpty(mDominantFontStyle)) {
+            if (mDominantFontStyle!!.isEmpty()) {
```
Comment (criticalAY) [inline AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ReviewerCustomFonts.kt:81]: Why `!!` ?

File: AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ReviewerCustomFonts.kt [hunk 3]
```diff
@@ -79,9 +78,9 @@ class ReviewerCustomFonts(context: Context) {
...
                 mDominantFontStyle = getDefaultFontStyle(context, customFontsMap)
-                if (TextUtils.isEmpty(mDominantFontStyle)) {
+                if (mDominantFontStyle!!.isEmpty()) {
```
Comment (criticalAY) [inline AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ReviewerCustomFonts.kt:83]: Same

File: AnkiDroid/src/main/java/com/ichi2/libanki/Finder.kt [hunk 4]
```diff
@@ -218,7 +217,7 @@ class Finder(private val col: Collection) {
             // failed command?
             @Suppress("NAME_SHADOWING")
             var txt = txt
-            if (TextUtils.isEmpty(txt)) {
+            if (txt.isNullOrEmpty()) {
```
Comment (criticalAY) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Finder.kt:220]: `isEmpty` is more suited here

File: annotations/README.md [hunk 5]
```diff
@@ -39,8 +39,8 @@ Sample:
         <annotationsPath>
           <root type="composite">
             <root url="jar://$USER_HOME$/AppData/Local/Android/Sdk/platforms/android-28/data/annotations.zip!/" type="simple" />
-            <root url="file://C:/GitHub/Anki-Android-David/annotations" type="simple" />
-```
+              <root url="file://C:/GitHub/Anki-Android-David/annotations" type="simple"/>
+```xml
```
Comment (criticalAY) [inline annotations/README.md:43]: Why this?

File: AnkiDroid/src/main/java/com/ichi2/utils/LanguageUtil.kt [hunk 6]
```diff
@@ -214,16 +213,16 @@ object LanguageUtil {
...
             // If no code provided use the app language.
         }
-        if (TextUtils.isEmpty(tempLocaleCode)) {
+        if (tempLocaleCode.isNullOrEmpty()) {
             // Fall back to (system) default only if that fails.
             tempLocaleCode = Locale.getDefault().toString()
         }
         // Language separators are '_' or '-' at different times in display/resource fetch
-        val locale: Locale = if (tempLocaleCode != null && (tempLocaleCode.contains("_") || tempLocaleCode.contains("-"))) {
+        val locale: Locale = if (tempLocaleCode.contains("_") || tempLocaleCode.contains("-")) {
```
Comment (criticalAY) [inline AnkiDroid/src/main/java/com/ichi2/utils/LanguageUtil.kt:225]: I am not in the favour of removing this condition from here

**Review by criticalAY — COMMENTED** [review]
I reviewed most of the files but not all waiting for a response

**Comment by criticalAY** [comment]
You marked resolved but are they?

**Comment by criticalAY** [comment]
You still haven't took care of few things that I mentioned.

**Review by BrayanDSO — CHANGES_REQUESTED** [review]
Tests fail and it's better to use `.isNullOrEmpty()` instead of `?.isEmpty() == true`

**Comment by soCallmeAdityaKumar** [comment]
> Tests fail and it's better to use `.isNullOrEmpty()` instead of `?.isEmpty() == true`

ok i will do it right now

**Comment by soCallmeAdityaKumar** [comment]
@BrayanDSO @criticalAY  I think my checks are failing because of the ankiAndroidBackend file that I have not install( placed correctly)  ( I have windows OS). Can you please assist me in doing that.

**Comment by BrayanDSO** [comment]
You can see the failed tests log by clicking on `Details` by the tests side here on GitHub.

There, it shows the following failed test:
```
FinderTest > test_findCards FAILED
    java.lang.AssertionError: expected:<5> but was:<0>
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:120)
        at org.junit.Assert.assertEquals(Assert.java:146)
        at com.ichi2.libanki.FinderTest.test_findCards(FinderTest.kt:276)
```
As this is an error on `FinderTest.kt`, which tests `Finder.kt`, and you've modified `Finder.kt`, probably the error is somewhere there, so you should try fixing it.

---
About your problem with the backend installation, I can't tell exactly what you mean, so I suggest reading [this article](https://stackoverflow.com/help/how-to-ask) about how to ask better a question on the internet and going to AnkiDroid Discord so someone can answer you better

**Comment by dae** [comment]
https://github.com/ankidroid/Anki-Android/issues/8482#issuecomment-1100924517

**Comment by soCallmeAdityaKumar** [comment]
> You can see the failed tests log by clicking on `Details` by the tests side here on GitHub.
> 
> There, it shows the following failed test:
> 
> ```
> FinderTest > test_findCards FAILED
>     java.lang.AssertionError: expected:<5> but was:<0>
>         at org.junit.Assert.fail(Assert.java:89)
>         at org.junit.Assert.failNotEquals(Assert.java:835)
>         at org.junit.Assert.assertEquals(Assert.java:120)
>         at org.junit.Assert.assertEquals(Assert.java:146)
>         at com.ichi2.libanki.FinderTest.test_findCards(FinderTest.kt:276)
> ```
> 
> As this is an error on `FinderTest.kt`, which tests `Finder.kt`, and you've modified `Finder.kt`, probably the error is somewhere there, so you should try fixing it.
> 
> About your problem with the backend installation, I can't tell exactly what you mean, so I suggest reading [this article](https://stackoverflow.com/help/how-to-ask) about how to ask better a question on the internet and going to AnkiDroid Discord so someone can answer you better

Ok I am looking over it.

**Comment by soCallmeAdityaKumar** [comment]
> [#8482 (comment)](https://github.com/ankidroid/Anki-Android/issues/8482#issuecomment-1100924517)

Thanks

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/Anki2Importer.kt [hunk 7]
```diff
@@ -421,19 +420,19 @@ open class Anki2Importer(col: Collection?, file: String) : Importer(col!!, file)
...
         // if there's a prefix, replace the top level deck
-        if (!TextUtils.isEmpty(mDeckPrefix)) {
+        if (!mDeckPrefix.isNullOrEmpty()) {
             val parts = listOf(*Decks.path(name))
-            val tmpname = TextUtils.join("::", parts.subList(1, parts.size))
-            name = mDeckPrefix!!
-            if (!TextUtils.isEmpty(tmpname)) {
+            val tmpname = parts.subList(1, parts.size).joinToString("::")
+            name = mDeckPrefix
+            if (!tmpname.isEmpty()) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/libanki/importer/Anki2Importer.kt:427]: The IDE actually suggests this
```suggestion
            if (tmpname.isNotEmpty()) {
```

File: AnkiDroid/src/main/java/com/ichi2/libanki/Finder.kt [hunk 8]
```diff
@@ -744,13 +743,13 @@ class Finder(private val col: Collection) {
                 "select c.id from cards c, notes n where c.nid=n.id and "
             }
             // combine with preds
-            sql += if (!TextUtils.isEmpty(preds)) {
+            sql += if (!preds.isEmpty()) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Finder.kt:746]: remove the `!` and use `isNotEmpty()`, as the IDE suggests

File: AnkiDroid/src/main/java/com/ichi2/libanki/Finder.kt [hunk 9]
```diff
@@ -744,13 +743,13 @@ class Finder(private val col: Collection) {
...
                 "($preds)"
             } else {
                 "1"
             }
             // order
-            if (!TextUtils.isEmpty(order)) {
+            if (!order.isEmpty()) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Finder.kt:752]: ditto

File: AnkiDroid/src/main/java/com/ichi2/utils/LanguageUtil.kt [hunk 10]
```diff
@@ -214,16 +213,16 @@ object LanguageUtil {
      */
     fun getLocale(localeCode: String?, prefs: SharedPreferences): Locale {
         var tempLocaleCode = localeCode
-        if (tempLocaleCode == null || TextUtils.isEmpty(tempLocaleCode)) {
+        if (tempLocaleCode == null || tempLocaleCode.isEmpty()) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/utils/LanguageUtil.kt:216]:
```suggestion
        if (tempLocaleCode.isNullOrEmpty()) {
```

File: AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt [hunk 11]
```diff
@@ -775,7 +774,7 @@ create table meta (dirMod int, lastUsn int); insert into meta values (0, 0);"""
                         val csum = cur.getString(1)
                         fnames.add(fname)
                         val normName = Utils.nfcNormalized(fname)
-                        if (!TextUtils.isEmpty(csum)) {
+                        if (!csum.isEmpty()) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt:777]: remove the `!` and use `isNotEmpty()`, as the IDE suggests

File: AnkiDroid/src/main/java/com/ichi2/libanki/Model.kt [hunk 12]
```diff
@@ -91,7 +90,7 @@ class Model : JSONObject {
         val fieldNames = fieldsNames
         val nonemptyFields: MutableSet<String> = HashUtil.HashSetInit(sfld.size)
         for (i in sfld.indices) {
-            if (!TextUtils.isEmpty(sfld[i].trim { it <= ' ' })) {
+            if (!sfld[i].trim { it <= ' ' }.isEmpty()) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Model.kt:93]: remove the `!` and use `isNotEmpty()`, as the IDE suggests

File: annotations/README.md [hunk 13]
```diff
@@ -31,17 +31,16 @@ Annotations should now appear
...
+    <version value="java version &quot;1.8.0_112-release&quot;" />
+    <homePath value="C:\Users\David\AppData\Local\Android\Sdk" />
+    <roots>
         <annotationsPath>
           <root type="composite">
-            <root url="jar://$USER_HOME$/AppData/Local/Android/Sdk/platforms/android-28/data/annotations.zip!/" type="simple" />
-            <root url="file://C:/GitHub/Anki-Android-David/annotations" type="simple" />
+             <root url="jar://$USER_HOME$/AppData/Local/Android/Sdk/platforms/android-28/data/annotations.zip!/" type="simple"/>
+             <root url="file://C:/GitHub/Anki-Android-David/annotations" type="simple"/>
 ```
```
Comment (BrayanDSO) [inline annotations/README.md:43]: Unnecessary/Unrelated changes. Please revert

File: AnkiDroid/src/main/java/com/ichi2/libanki/sync/RemoteMediaServer.kt [hunk 14]
```diff
@@ -156,7 +155,7 @@ class RemoteMediaServer(
     @Throws(MediaSyncException::class)
     // NOTE: the original name of the method was _dataOnly which followed upstream naming
     private inline fun <reified T> dataOnly(resp: JSONObject, returnType: Class<T>): T {
-        if (!TextUtils.isEmpty(resp.optString("err"))) {
+        if (!resp.optString("err").isEmpty()) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sync/RemoteMediaServer.kt:158]: remove the `!` and use `isNotEmpty()`, as the IDE suggests

File: AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ReviewerCustomFonts.kt [hunk 15]
```diff
@@ -79,9 +78,9 @@ class ReviewerCustomFonts(context: Context) {
     private fun getDominantFontStyle(context: Context, customFontsMap: Map<String?, AnkiFont>): String? {
         if (mDominantFontStyle == null) {
             mDominantFontStyle = getOverrideFontStyle(context, customFontsMap)
-            if (TextUtils.isEmpty(mDominantFontStyle)) {
+            if (mDominantFontStyle?.isEmpty() == true) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ReviewerCustomFonts.kt:81]:
```suggestion
            if (mDominantFontStyle.isNullOrEmpty()) {
```

File: AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ReviewerCustomFonts.kt [hunk 16]
```diff
@@ -79,9 +78,9 @@ class ReviewerCustomFonts(context: Context) {
...
                 mDominantFontStyle = getDefaultFontStyle(context, customFontsMap)
-                if (TextUtils.isEmpty(mDominantFontStyle)) {
+                if (mDominantFontStyle?.isEmpty() == true) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ReviewerCustomFonts.kt:83]:
```suggestion
                if (mDominantFontStyle.isNullOrEmpty()) {
```

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/Sched.kt [hunk 17]
```diff
@@ -853,7 +852,7 @@ class Sched(col: Collection) : SchedV2(col) {
         val limit = terms.getInt(1)
         val order = terms.getInt(2)
         val orderLimit: SortOrder = AfterSqlOrderBy(_dynOrder(order, limit))
-        if (!TextUtils.isEmpty(search.trim { it <= ' ' })) {
+        if (!search.trim { it <= ' ' }.isEmpty()) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sched/Sched.kt:855]: remove the `!` and use `isNotEmpty()`, as the IDE suggests

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.kt [hunk 18]
```diff
@@ -1778,7 +1777,7 @@ open class SchedV2(col: Collection) : AbstractSched(col) {
             val limit = term.getInt(1)
             val order = term.getInt(2)
             val orderLimit = _dynOrder(order, limit)
-            if (!TextUtils.isEmpty(search.trim { it <= ' ' })) {
+            if (!search.trim { it <= ' ' }.isEmpty()) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.kt:1780]: remove the `!` and use `isNotEmpty()`, as the IDE suggests

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/Shared.kt [hunk 19]
```diff
@@ -55,7 +54,7 @@ object Shared {
      * @return See getTestDir.
      */
     private fun getTestDir(context: Context, name: String): File {
-        val suffix = if (!TextUtils.isEmpty(name)) {
+        val suffix = if (!name.isEmpty()) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/Shared.kt:57]: remove the `!` and use `isNotEmpty()`, as the IDE suggests

File: AnkiDroid/src/main/java/com/ichi2/libanki/stats/Stats.kt [hunk 20]
```diff
@@ -216,7 +215,7 @@ class Stats(private val col: com.ichi2.libanki.Collection, did: Long) {
 
     private fun _deckAge(by: DeckAgeType): Long {
         var lim = _revlogLimit()
-        if (!TextUtils.isEmpty(lim)) {
+        if (!lim.isEmpty()) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/libanki/stats/Stats.kt:218]: remove the `!` and use `isNotEmpty()`, as the IDE suggests

File: AnkiDroid/src/main/java/com/ichi2/libanki/stats/Stats.kt [hunk 21]
```diff
@@ -246,7 +245,7 @@ class Stats(private val col: com.ichi2.libanki.Collection, did: Long) {
     private fun getRevlogFilter(timespan: AxisType, inverseTimeSpan: Boolean): String {
         val lims = ArrayList<String>(2)
         val dayFilter = getRevlogTimeFilter(timespan, inverseTimeSpan)
-        if (!TextUtils.isEmpty(dayFilter)) {
+        if (!dayFilter.isEmpty()) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/libanki/stats/Stats.kt:248]: remove the `!` and use `isNotEmpty()`, as the IDE suggests

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsDialog.kt [hunk 22]
```diff
@@ -193,7 +192,7 @@ class TagsDialog : AnalyticsDialogFragment {
         val toolbarAddItem = toolbar.menu.findItem(R.id.tags_dialog_action_add)
         toolbarAddItem.setOnMenuItemClickListener {
             val query = mToolbarSearchView!!.query.toString()
-            if (mToolbarSearchItem!!.isActionViewExpanded && !TextUtils.isEmpty(query)) {
+            if (mToolbarSearchItem!!.isActionViewExpanded && !query.isEmpty()) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsDialog.kt:195]: remove the `!` and use `isNotEmpty()`, as the IDE suggests

File: AnkiDroid/src/main/java/com/ichi2/libanki/template/TemplateFilters.kt [hunk 23]
```diff
@@ -69,7 +68,7 @@ object TemplateFilters {
         var filter = filterInput
         return if ("text" == filter) {
             // strip html
-            if (!TextUtils.isEmpty(txt)) {
+            if (!txt.isEmpty()) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/libanki/template/TemplateFilters.kt:71]: remove the `!` and use `isNotEmpty()`, as the IDE suggests

File: AnkiDroid/src/main/java/com/ichi2/libanki/template/TemplateFilters.kt [hunk 24]
```diff
@@ -83,7 +82,7 @@ object TemplateFilters {
             val split = filter.split("-").toTypedArray()
             filter = split[0]
             val extra = split[1]
-            if (!TextUtils.isEmpty(txt) && !TextUtils.isEmpty(extra)) {
+            if (!txt.isEmpty() && !extra.isEmpty()) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/libanki/template/TemplateFilters.kt:85]: remove the `!` and use `isNotEmpty()`, as the IDE suggests

File: AnkiDroid/src/main/java/com/ichi2/libanki/template/TemplateFilters.kt [hunk 25]
```diff
@@ -137,7 +136,7 @@ object TemplateFilters {
             @KotlinCleanup("maybe make non-null")
             var buf: String?
             buf = if (type == 'q') {
-                if (!TextUtils.isEmpty(m.group(4))) {
+                if (!m.group(4).isNullOrEmpty()) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/libanki/template/TemplateFilters.kt:139]: remove the `!` and use `isNotEmpty()`, as the IDE suggests

File: AnkiDroid/src/main/java/com/ichi2/libanki/Utils.kt [hunk 26]
```diff
@@ -1124,7 +1123,7 @@ object Utils {
         for (kv in fields.entries) {
             var value = kv.value
             value = stripHTMLMedia(value).trim { it <= ' ' }
-            if (!TextUtils.isEmpty(value)) {
+            if (!value.isEmpty()) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Utils.kt:1126]: remove the `!` and use `isNotEmpty()`, as the IDE suggests

**Review by BrayanDSO — CHANGES_REQUESTED** [review]
Overall, the PR seems pretty good. Thanks for the job!

The requested changes are more about avoiding leaving IDE lints.

Another request that I do is to remove the last two usages of `TextUtils.isEmpty()` on `DeckOptionsActivity.kt`, so we can close the issue

**Comment by BrayanDSO** [comment]
I'm not sure if you want another review, @soCallmeAdityaKumar, buf if you do, there are some unresolved requests to do first

**Comment by soCallmeAdityaKumar** [comment]
> Overall, the PR seems pretty good. Thanks for the job!
> 
> The requested changes are more about avoiding leaving IDE lints.
> 
> Another request that I do is to remove the last two usages of `TextUtils.isEmpty()` on `DeckOptionsActivity.kt`, so we can close the issue

I have done this in DeckOptionsActivity

**Comment by soCallmeAdityaKumar** [comment]
@BrayanDSO  i have addressed all your requested changes

File: AnkiDroid/src/main/java/com/ichi2/libanki/Model.kt (hunk 12 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Model.kt:93]: Hasn't been resolved

File: AnkiDroid/src/main/java/com/ichi2/libanki/sync/RemoteMediaServer.kt (hunk 14 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sync/RemoteMediaServer.kt:158]: Hasn't been resolved

File: AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ReviewerCustomFonts.kt (hunk 15 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ReviewerCustomFonts.kt:81]: Hasn't been resolved

File: AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ReviewerCustomFonts.kt (hunk 16 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ReviewerCustomFonts.kt:83]: Hasn't been resolved

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/Sched.kt (hunk 17 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sched/Sched.kt:855]: Hasn't been resolved

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.kt (hunk 18 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.kt:1780]: Hasn't been resolved

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/Shared.kt (hunk 19 above)
Comment (BrayanDSO) [reply AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/Shared.kt:57]: Hasn't been resolved

File: AnkiDroid/src/main/java/com/ichi2/libanki/stats/Stats.kt (hunk 20 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/libanki/stats/Stats.kt:218]: Hasn't been resolved

File: AnkiDroid/src/main/java/com/ichi2/libanki/stats/Stats.kt (hunk 21 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/libanki/stats/Stats.kt:248]: Hasn't been resolved

**Comment by BrayanDSO** [comment]
@soCallmeAdityaKumar, I commented with `Hasn't been resolved` the changes that haven't been resolved. Github hides some requests when there are many, so make sure to unhide them

**Review by BrayanDSO — APPROVED** [review]
LGTM. Thanks!

**Review by criticalAY — APPROVED** [review]

**Comment by BrayanDSO** [comment]
We have the second approval of @criticalAY, so we are good to go

**Pull request merged**